rustls/src/hash_hs.rs

```diff
@@ -15,10 +15,7 @@ use std::mem;
 /// This is disabled in cases where client auth is not possible.
 pub struct HandshakeHash {
     /// None before we know what hash function we're using
-    alg: Option<&'static digest::Algorithm>,
-
-    /// None before we know what hash function we're using
-    ctx: Option<digest::Context>,
+    ctx: Option<(&'static digest::Algorithm, digest::Context)>,
```

Comment:
`digest::Context::algorithm()` provides access to the algorithm so you should remove the first part of this tuple.